SneakPeek/src/com/cyscorpions/dalejulian/sneakpeek/adapters/SneakerCategoryAdapter.java

```diff
@@ -0,0 +1,62 @@
+package com.cyscorpions.dalejulian.sneakpeek.adapters;
+
+import java.util.ArrayList;
+
+import com.cyscorpions.dalejulian.sneakpeek.R;
+import com.cyscorpions.dalejulian.sneakpeek.R.id;
+import com.cyscorpions.dalejulian.sneakpeek.models.SneakerCategory;
+
+import android.app.Activity;
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+
+public class SneakerCategoryAdapter extends ArrayAdapter<SneakerCategory> {
+	private Context mContext;
+	private int layoutResourceId;
+	// SneakerCategory[] data = null;
+	private ArrayList<SneakerCategory> data;
+
+	// view holder
+	static class CategoryHolder {
+		TextView categoryName, categoryDesc;
+	}
+
+	public SneakerCategoryAdapter(Context appContext, int layoutResourceId,
+			ArrayList<SneakerCategory> data) {
+		super(appContext, layoutResourceId, data);
+		this.mContext = appContext;
+		this.layoutResourceId = layoutResourceId;
```

Comment:
I don't think you need to pass a layout resource id into the adapter. You can just use the layout resource id inside the getView method since this adapter will specifically be using the same layout resource if ever it will be reused